### ISSUE TITLE
A question about outputs.logits in caption.py

### ISSUE THREAD

**Issue by AaronNZH** [Open3DA/LL3DA#8]
detector_output['loss'] += self.loss_caption(
            logits = outputs.logits[:, prefix_tokens.shape[1] - 1: -1],
            target = input_ids,
            mask = gradient_mask.to(self.dtype),
        )

I'm not familiar with huggingface transformers. Why does line 342 in caption.py using logits = **outputs.logits[:, prefix_tokens.shape[1] - 1: -1]**, rather than logits = **outputs.logits[:, prefix_tokens.shape[1]:]**. It seems to include the last learnable query in qformer while dropping the last word embedding.

**Comment by ch3cook-fdu** [comment]
Yes, this is a common practice as we are training the model to **generate the $i$-th token given the previous $i-1$ tokens**. 

Please see the following implementations for more details:

https://github.com/huggingface/transformers/blob/main/src/transformers/models/opt/modeling_opt.py#L1146-L1151

https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt2/modeling_gpt2.py#L1102-L1107

**Comment by AaronNZH** [comment]
Got it, thank you for your explanation!
